Add support for never keyword

jaapio opened the pull request:
php 8.1 introduces the `never` keyword, which can be used as return type.

jaapio committed:
Add support for never keyword

php 8.1 introduces the `never` keyword, which can be used as return type.

mvriel commented on src/Types/Never_.php:
```diff
@@ -0,0 +1,35 @@
+use phpDocumentor\Reflection\Type;
+
+/**
+ * Value Object representing the pseudo-type 'void'.
```
Please update this description

mvriel replied:
Also, never is not a pseudo-type, right? It will be an actual type?

jaapio replied:
the docs of php are not updated yet with this information... I'm not sure if it will be an actual type.. since you cannot used it like other scalar types. it is only allowed for return types.

ashnazg commented:
Yes, `never` is a native type.
CRB

On Fri, Sep 17, 2021, 10:11 Mike van Riel ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/Types/Never_.php
> <https://github.com/phpDocumentor/TypeResolver/pull/130#discussion_r711141433>
> :
>
> > +
> +/**
> + * This file is part of phpDocumentor.
> + *
> + * For the full copyright and license information, please view the LICENSE
> + * file that was distributed with this source code.
> + *
> + * @link      http://phpdoc.org
> + */
> +
> +namespace phpDocumentor\Reflection\Types;
> +
> +use phpDocumentor\Reflection\Type;
> +
> +/**
> + * Value Object representing the pseudo-type 'void'.
>
> Also, never is not a pseudo-type, right? It will be an actual type?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/phpDocumentor/TypeResolver/pull/130#discussion_r711141433>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAYOSQ5EPEIP7E3CTGMIMLUCNLC3ANCNFSM5EHQMM4A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

jaapio committed:
Improve docs about return only types

Void and never are types that can only be used for function return
types. Officially php doesn't have a category of types that are only
allowed for return, but I think it makes sense to name them like this.

jaapio commented:
I named it `return-type` because that fits the purpose :-)
